src/util/merge-queries.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import type { Location, Query } from '../types';
+
+import qs from 'query-string';
+
+export default (oldQuery: Query = {}, newQuery: Query = {}): Location => {
```

Comment:
We can avoid the default params in this case, object spread works fine with `undefined`